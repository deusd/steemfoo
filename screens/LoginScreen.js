import React from 'react'
import { Alert, Text, StyleSheet, TouchableOpacity } from 'react-native'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import PageContainer from '../components/PageContainer'
import Logo from '../components/Logo'
import { login } from '../state/steem'

export class LoginScreen extends React.Component {
  static propTypes = {
    login: PropTypes.func.isRequired,
  }

  render() {
    if (this.props.error) {
      Alert.alert('Oopsy', 'There was an error signing in, please try again')
    }

    return (
      <PageContainer style={styles.page}>
        <Logo style={styles.logo} />
        <Text style={styles.text}>
          All account operations are handled securely through steemconnect
        </Text>
        <Text style={styles.smallText}>
          We never store your passwords or private keys locally
        </Text>
        <TouchableOpacity
          style={styles.button}
          onPress={this.props.login}
          style={styles.button}
        >
          <Text style={styles.buttonText}>Login</Text>
        </TouchableOpacity>
      </PageContainer>
    )
  }
}

const styles = StyleSheet.create({
  page: {
    backgroundColor: '#3BB8FF',
    padding: 20,
  },
  text: {
    fontSize: 18,
    fontFamily: 'Roboto-Regular',
    color: 'white',
    textAlign: 'center',
    marginTop: 30,
  },
  smallText: {
    fontSize: 14,
    fontFamily: 'Roboto-Regular',
    color: 'white',
    textAlign: 'center',
    marginTop: 10,
  },
  logo: {
    marginTop: 120,
    alignSelf: 'center',
  },
  button: {
    borderWidth: 3,
    borderColor: 'white',
    borderRadius: 5,
    marginTop: 40,
    padding: 10,
    width: '100%',
    flexDirection: 'row',
    justifyContent: 'center',
  },
  buttonText: {
    color: 'white',
    fontFamily: 'Roboto-Regular',
    fontSize: 18,
  },
})

export default connect(
  ({ steem }) => ({
    user: steem.user,
    error: steem.signinError,
  }),
  { login }
)(LoginScreen)
