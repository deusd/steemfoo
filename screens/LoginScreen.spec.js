import React from 'react'
import { Alert } from 'react-native'
import { shallow } from 'enzyme'
import { LoginScreen } from './LoginScreen'

describe('LoginScreen', () => {
  let wrapper
  let login

  beforeEach(() => {
    login = jest.fn()
    wrapper = shallow(<LoginScreen login={login} />)
    jest.spyOn(Alert, 'alert')
  })

  it('should render', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should call the login function on pressing the login button', () => {
    wrapper.setProps({ login })
    const button = wrapper.find('TouchableOpacity').first()
    button.props().onPress()
    expect(login).toHaveBeenCalledTimes(1)
  })

  it('should show the alert when there is an error', () => {
    wrapper.setProps({ error: true })
    expect(Alert.alert).toHaveBeenCalledTimes(1)
  })
})
