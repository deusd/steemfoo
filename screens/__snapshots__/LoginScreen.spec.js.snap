// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginScreen should render 1`] = `
<PageContainer
  style={
    Object {
      "backgroundColor": "#3BB8FF",
      "padding": 20,
    }
  }
>
  <Logo
    style={
      Object {
        "alignSelf": "center",
        "marginTop": 120,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "white",
        "fontFamily": "Roboto-Regular",
        "fontSize": 18,
        "marginTop": 30,
        "textAlign": "center",
      }
    }
  >
    All account operations are handled securely through steemconnect
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "white",
        "fontFamily": "Roboto-Regular",
        "fontSize": 14,
        "marginTop": 10,
        "textAlign": "center",
      }
    }
  >
    We never store your passwords or private keys locally
  </Text>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[MockFunction]}
    style={
      Object {
        "borderColor": "white",
        "borderRadius": 5,
        "borderWidth": 3,
        "flexDirection": "row",
        "justifyContent": "center",
        "marginTop": 40,
        "padding": 10,
        "width": "100%",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "fontFamily": "Roboto-Regular",
          "fontSize": 18,
        }
      }
    >
      Login
    </Text>
  </TouchableOpacity>
</PageContainer>
`;
